test/mpi/test_allgatherv.cc

```diff
@@ -65,8 +65,13 @@ TestAllgatherv::TestAllgatherv(ucc_test_team_t &_team, TestCaseParams &params) :
         UCC_CHECK(ucc_mc_alloc(&sbuf_mc_header, counts[rank] * dt_size, mem_type));
         sbuf = sbuf_mc_header->addr;
     } else {
-        args.mask = UCC_COLL_ARGS_FIELD_FLAGS;
-        args.flags = UCC_COLL_ARGS_FLAG_IN_PLACE;
+        args.mask  |= UCC_COLL_ARGS_FIELD_FLAGS;
+        args.flags |= UCC_COLL_ARGS_FLAG_IN_PLACE;
+    }
+
+    if (persistent) {
```

Comment:
i think it's better to set UCC_COLL_ARGS_FLAG_PERSISTENT in TestCase ctor, then you don't need to copy paste same code for each collective subclass. Also do the same for inplace flag

Reply:
Will move for persistent.
Regarding inplace though, as it is not supported in bcast and barrier, setting flags in ctor will be problematic. I can skip test in inplace case in bcast and barrier, or that the inplace flag needs to be set in each coll seperately as is now. What do you think @Sergei-Lebedev?

Reply:
@manjugv @vspetrov if we set UCC_COLL_ARGS_FLAG_IN_PLACE for collectives that don't support inplace mode such as barrier what is the expected behavior? is it an error i.e. invalid argument or for such collective inplace flag simply ignored?

Reply:
It is user responsibility to clear/init coll_args->mask. So, my take is, if user sets UCC_COLL_ARGS_FIELD_FLAGS as mask bit, then user must initialize "flags" properly. And if user put IN_PLACE for bcast in that case, it is invalid param. We could just ignore that of course, but i think good implementation should print error. Because, this means user set the flag on purpose (since both mask value is set and flag is set) - so user must understand that he is not getting what he wants.

Reply:
Agree @vspetrov. We can print warning instead of error though.